src/gui/mzroll/background_peaks_update.cpp

```diff
@@ -587,68 +571,49 @@ void BackgroundPeakUpdate::align() {
                         mainwindow->alignmentPolyVizDockWidget->setCoefficientMap(aligner.sampleCoefficient);
                 } else if (alignAlgo == 1) {
                         aligner.preProcessing(groups);
-
+                        // initialize processedDataFromPython with null 
+                        processedDataFromPython="";
                          /**runPythonProg()
                          * sends the json of groups and samples rt to the python exe. for more look in sendDataToPython()
                          * python exe is going to correct the rts and send it back to us in json format
                          */
                         runPythonProg(&aligner);
-
-                        /**readDataFromPython(data)
-                         * it will wait for python to send the corrected rts
-                         * once we receive the corrected rts, we kill the python exe and update the rts in maven
-                         */
-                        QByteArray data;
-                        readDataFromPython(data);
+                        // wait for processing of data by python program
+                        pythonProg->waitForFinished(-1);
```

Comment:
Why not use [QProcess::finshed()](https://doc.qt.io/qt-5/qprocess.html#finished) to handle things after the process has been completed. We can send proper exit status from python and  use it here  to check if python algo failed/succeeded. 
What do you think ?